services/wireguard/service/service_unix.go

```diff
@@ -127,3 +145,12 @@ func (manager *Manager) Stop() error {
 	log.Info("wireguard service stopped")
 	return nil
 }
+
+func providerIP(subnet net.IPNet) net.IP {
```

Comment:
Looks very complicated function, I suppose we already had one in this package

Reply:
We have similar functions in a number of places, but all of them a bit different. We probably need some general and complex function that will work for all cases. I didn't want to add it right now.

Reply:
Added this function to the utils package as a single place.